aiohttp/web_urldispatcher.py

```diff
@@ -997,21 +1009,56 @@ class UrlDispatcher(AbstractRouter, Mapping[str, AbstractResource]):
 
     def __init__(self) -> None:
         super().__init__()
-        self._resources: List[AbstractResource] = []
-        self._named_resources: Dict[str, AbstractResource] = {}
+        self._resources: list[AbstractResource] = []
+        self._named_resources: dict[str, AbstractResource] = {}
         self._resource_index: dict[str, list[AbstractResource]] = {}
-        self._matched_sub_app_resources: List[MatchedSubAppResource] = []
+        self._matched_sub_app_resources: list[MatchedSubAppResource] = []
+        self._hyperdb: hyperscan.Database | None = None  # type: ignore[no-any-unimported]
+        self._hyper_index: dict[int, AbstractResource] = {}
+
+    async def _resolve_fast(
+        self, request: Request, path: str, allowed_methods: set[str]
+    ) -> UrlMappingMatchInfo | None:
+        if self._hyperdb is None:
+            return await self._resolve_fallback(request, path, allowed_methods)
+
+        # N.B.: allowed_methods variable in modified in-place
+        found: list[int] = []
+
+        def on_match(
+            id_: int, from_: int, to: int, flags: int, context: Any | None
+        ) -> bool | None:
+            found.append(id_)
+            return None
+
+        self._hyperdb.scan(path.encode("utf8"), match_event_handler=on_match)
```

Comment:
This is really nice. Didn't know about hyperscan before and could have used it to solve other problems in the past.  Thanks for turning me on to it